### DISCUSSION TITLE
TreeDataGrid: How to set first row "IsSelected = true"

### DISCUSSION THREAD

**Discussion by bsdRuf** [AvaloniaUI/Avalonia#16142] [Q&A] [answered]
Hello,

i am building my TDG dynamicly. By creating custom FileTreeNodeModels. I tried to set the IsSelected in there but that didnt work.

```
TreeDataGridRow.IsSelected = true;
```

But TreeDataGridRow.IsSelected is only a getter. From the F12 I could see there is a CLR Prop which I could set in the F12 GUI but I dont know how to set it by the constructor of the window.

Simply its just to select/highlight the first entry of my TDG (its used as navigation - somehow like the windows 11 views)
Any idea or advice is highly appreciated

![image](https://github.com/AvaloniaUI/Avalonia/assets/136796815/a322c06e-6c48-4b54-8d93-d788fdf668f2)

**Comment by timunie** [comment, marked as answer]
myTreeDataGrid.RowSelection.Select(idx) iirc

**Reply by bsdRuf** [reply]
Your memory is awsome, thank you
